backend/src/migrations/008_delete_reviews_with_removed_access_requests.ts

```diff
@@ -0,0 +1,21 @@
+import AccessRequestModel from '../models/AccessRequest.js'
+import ReviewModel from '../models/Review.js'
+import { removeAccessRequestReviews } from '../services/review.js'
+
+export async function up() {
+  const accessRequests = await AccessRequestModel.find({ deleted: false })
+  const reviews = await ReviewModel.find({})
```

Comment:
Why do you need to iterate though every review (there could be a lot)? Surely once you have all of the deleted access requests you just need to call `removeAccessRequestReviews()` for each of them with the `accessRequestID`.

Reply:
The mongoose-delete plugin doesn't work nicely in typescript. Ideally, I'd make `AccessRequestModel` a `SoftDeleteModel` and use the `findDeleted` method but this doesn't seem to work as expected, even with `npm install --save @types/mongoose-delete` helping out. Instead, we have to iterate over all of the reviews and find which ones are "orphaned" by not having an associated access request, and removing them. It would be worth revisiting this at another time to get this missing functionality as it could be useful for future use.

Reply:
Using the not-best-practice `(AccessRequestModel as any).findDeleted()` has allowed for a workaround. This should really be sorted for any future work.